modules/core/shared/src/test/scala/freestyle/tagless/TaglessTests.scala

```diff
@@ -34,6 +34,107 @@ import modules._
 
 class TaglessTests extends WordSpec with Matchers {
 
+  "the @tagless macro annotation should be accepted if it is applied to" when {
+
+    "a trait with at least one request" in {
+      "@tagless trait X { def bar(x:Int): FS[Int] }" should compile
+    }
+
+    "a trait with an F-Bound type param" in {
+      "@tagless trait FBound[F[_]] { def bar(x:Int): FS[Int] }" should compile
```

Comment:
We should probably also support `F[Int]` since in this case it's the same as `FS[Int]`
```scala
@tagless trait FBound[F[_]] { def bar(x:Int): F[Int] }
```